src/rt/test/func/memory/memory_iterator.h

```diff
@@ -280,14 +280,107 @@ namespace memory_iterator
       snmalloc::debug_check_empty<snmalloc::Alloc::StateHandle>();
       check(live_count == 0);
     }
+    else if constexpr (region_type == RegionType::Rc)
+    {
+      static constexpr uintptr_t FINALISER_MASK = 1 << 1;
+      using C = C3<region_type>;
+      using F = F3<region_type>;
+
+      C* oc = nullptr;
+      F* of = nullptr;
+      std::unordered_set<Object*> s1; // all objects
+      std::unordered_set<Object*> s2; // trivial
+      std::unordered_set<Object*> s3; // non-trivial
+      auto& alloc = ThreadAlloc::get();
+
+      auto* r = new (alloc) C;
+      auto* reg = RegionRc::get(r);
+      /* s1.insert(r); */
+      /* s2.insert(r); // Note: might need to change this if o's type is
+       * changed. */
+
+      // Add some objects.
+      oc = new (alloc, r) C;
+      s1.insert(oc);
+      s2.insert(oc);
+
```

Comment:
Don't we leak `oc` here, as we aren't handling the RC correctly?

Reply:
I'm not quite sure I understand. Really this particular test is orthogonal to RC and is just checking that the iterator goes through every object in the bag. I can manually create a link and add an incref for `oc`, but I don't think it will prevent leaks because the entire region gets released after the test anyway.